src/app/model/User.ts

```diff
@@ -27,6 +27,34 @@ const defaultUserData: UserInterface = {
 class User extends BaseModel {
   VolunteerStatus = VolunteerStatus;
 
+  /**
+   * Given a displayName returns an user id
+   * @param {string} displayName : displayName of user
+   * @return {string}
+   */
+  async getIdByDisplayName(displayName: string): Promise<string | undefined> {
+    let collection = this.getCollection("users");
+    let doc;
+    try {
+      doc = await collection.where("displayName", "==", displayName).get();
+    } catch (error) {
+      //TODO show error message to user
+      throw error;
+    }
+
+    if (doc.empty) {
+      throw Error(`This user: ${displayName} does not exist`);
+    }
+
+    doc.forEach((data) => {
+      if (data === undefined) {
+        throw Error(`no data for this user: ${displayName}`);
+      }
+    });
+
+    return doc.docs[0].id;
```

Comment:
I would propose returning the whole user object. That way this method is more useful.

Reply:
Only the first user object or the whole array? Right now the displayNames are not exclusive to each user so there could be multiple results...

Reply:
Ah. If that's the case, how do we decide which Id to return? Just the first one?

Reply:
I don't have any better idea in mind.

Reply:
If this works, this works! I have no problem.

Reply:
The only problem is the user input. I guess the problem will be resolved as soon as the user search component (PR #252) is merged as this should return an user id instead of the displayName.

Reply:
That's awesome! We can refactor this later then. Let's get this PR in to master. I will start my work after that :P